Assets/Scripts/View/PlayerCanvas/Cell.cs

```diff
@@ -45,21 +56,35 @@ protected override void SetUpgradeMode(bool owner)
         Upgrade.SetActive(tof);
     }
 
+    void SendMessage(byte id)
+    {
+        controller.SendFabricState(id, player.PhotonView.Owner.ActorNumber, pos, director.Money);
+    }
+
     public void BuyFabric()
     {
         if (director != null)
+        {
+            SendMessage(21);
```

Comment:
:) I wonder how do you keep track on what each message id means? I would propose to use an enum for it.